### PR TITLE
[Update Detekt Version] Update Detekt to 1.17.1

### LINKED ISSUE #16895: Update Detekt Version - Update to 1.17.1
Parent #16892

This issue is about updating Detekt to the next version, the [1.17.1](https://github.com/detekt/detekt/releases/tag/v1.17.1).

While updating Detekt to the next version, new rules might end up being enabled that would need to be resolved/suppressed as well. This issue will deal with that too.

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent: #16892
Closes: #16895

This PR:
- Updates Detekt to `1.17.1` (from `1.16.0`) (a6a009c5dbf2cdeaca15447ef99681f29794e390).
- Resolved/Suppresses new warnings that started appearing due to the fact the below test sources are now included by default (previously excluded by default)
    - src/test/java
    - src/test/kotlin

PS: I recommend reviewing this PR commit-by-commit.

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough (especially the `detekt` check).

Merge instructions:

- [x] Wait for #16927 to be approved and merged to `trunk`.
- [x] Make sure this PR's base is automatically updated to `trunk`.
- [x] Update PR with latest `trunk`.
- [x] Remove `[Status] Not Ready for Merge` label.
- [ ] Merge PR to `trunk`.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModelTest.kt [hunk 1]
```diff
@@ -743,7 +745,7 @@ class ReaderDiscoverViewModelTest {
     }
 
     private fun createReaderTagList(numOfTags: Int = 1) = ReaderTagList().apply {
-        for (x in 0 until numOfTags) {
+        for (it in 0 until numOfTags) {
```
Comment (oguzkocer) [inline WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModelTest.kt]:
```suggestion
        (0 until numOfTags).forEach {
```

File: WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilderTest.kt [hunk 2]
```diff
@@ -996,7 +998,7 @@ class ReaderPostUiStateBuilderTest {
     }
 
     private fun createReaderTagList(numOfTags: Int) = ReaderTagList().apply {
-        for (x in 0 until numOfTags) {
+        for (it in 0 until numOfTags) {
```
Comment (oguzkocer) [inline WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilderTest.kt]:
```suggestion
        (0 until numOfTags).forEach {
```

File: WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/utils/StatsSiteProviderTest.kt [hunk 3]
```diff
@@ -116,7 +116,7 @@ class StatsSiteProviderTest : BaseUnitTest() {
             event = it.getContentIfNotHandled()
         }
 
-        for (i in 0..2) {
+        for (it in 0..2) {
```
Comment (oguzkocer) [inline WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/utils/StatsSiteProviderTest.kt]:
```suggestion
        (0..2).forEach {
```

File: WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/utils/StatsSiteProviderTest.kt [hunk 4]
```diff
@@ -134,7 +134,7 @@ class StatsSiteProviderTest : BaseUnitTest() {
             event = it.getContentIfNotHandled()
         }
 
-        for (i in 0..3) {
+        for (it in 0..3) {
```
Comment (oguzkocer) [inline WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/utils/StatsSiteProviderTest.kt]:
```suggestion
        (0..3).forEach {
```

**Review by oguzkocer — APPROVED** [review]
@ParaskP7 I left a few syntax suggestions. Looks great, thank you for this update! 🙇‍♂️

**Comment by ParaskP7** [comment]
> @ParaskP7 I left a few syntax suggestions. Looks great, thank you for this update! 🙇‍♂️

Thank you so much for the review and the 👍 @oguzkocer , I applied all your syntax suggestions! 🙇 ❤️ 🚀

**Pull request merged**